doc/api/async_hooks.md

```diff
@@ -327,20 +335,11 @@ current Node.js instance.
 
 The `type` is a string identifying the type of resource that caused
 `init` to be called. Generally, it will correspond to the name of the
-resource's constructor.
-
-Valid values are:
-
-```text
-FSEVENTWRAP, FSREQCALLBACK, GETADDRINFOREQWRAP, GETNAMEINFOREQWRAP, HTTPINCOMINGMESSAGE,
-HTTPCLIENTREQUEST, JSSTREAM, PIPECONNECTWRAP, PIPEWRAP, PROCESSWRAP, QUERYWRAP,
-SHUTDOWNWRAP, SIGNALWRAP, STATWATCHER, TCPCONNECTWRAP, TCPSERVERWRAP, TCPWRAP,
-TTYWRAP, UDPSENDWRAP, UDPWRAP, WRITEWRAP, ZLIB, SSLCONNECTION, PBKDF2REQUEST,
-RANDOMBYTESREQUEST, TLSWRAP, Microtask, Timeout, Immediate, TickObject
-```
```

Comment:
I would leave the list but be clear that it's the current state and not promised to remain consistent. The type list is not so much of a risk as the resource itself, so this is fairly likely to remain if the API itself remains.

Reply:
Maybe keep just a part? The list tends to be updated incomplete and in the end it should not look like a spec here.

Reply:
I think the following note about AsyncResource producing more types can function to make it clear that it's not a comprehensive list.

Reply:
The list was already outdated (again) so I haven't it added back completely. Instead I added some samples and direct read to the source code for more.